pkg/services/folder/folderimpl/folder.go

```diff
@@ -343,31 +344,29 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	}
 
 	var nestedFolder *folder.Folder
-	if s.features.IsEnabled(featuremgmt.FlagNestedFolders) {
-		cmd := &folder.CreateFolderCommand{
-			// TODO: Today, if a UID isn't specified, the dashboard store
-			// generates a new UID. The new folder store will need to do this as
-			// well, but for now we take the UID from the newly created folder.
-			UID:         dash.UID,
-			OrgID:       cmd.OrgID,
-			Title:       cmd.Title,
-			Description: cmd.Description,
-			ParentUID:   cmd.ParentUID,
-		}
-		nestedFolder, err = s.nestedFolderCreate(ctx, cmd)
-		if err != nil {
-			// We'll log the error and also roll back the previously-created
-			// (legacy) folder.
-			logger.Error("error saving folder to nested folder store", "error", err)
-			// do not shallow create error if the legacy folder delete fails
-			if deleteErr := s.dashboardStore.DeleteDashboard(ctx, &dashboards.DeleteDashboardCommand{
-				ID:    createdFolder.ID,
-				OrgID: createdFolder.OrgID,
-			}); deleteErr != nil {
-				logger.Error("error deleting folder after failed save to nested folder store", "error", err)
-			}
-			return dashboards.FromDashboard(dash), err
+	cmd = &folder.CreateFolderCommand{
+		// TODO: Today, if a UID isn't specified, the dashboard store
+		// generates a new UID. The new folder store will need to do this as
+		// well, but for now we take the UID from the newly created folder.
+		UID:         dash.UID,
+		OrgID:       cmd.OrgID,
+		Title:       cmd.Title,
+		Description: cmd.Description,
+		ParentUID:   cmd.ParentUID,
+	}
+	nestedFolder, err = s.nestedFolderCreate(ctx, cmd)
+	if err != nil {
+		// We'll log the error and also roll back the previously-created
+		// (legacy) folder.
+		logger.Error("error saving folder to nested folder store", "error", err)
+		// do not shallow create error if the legacy folder delete fails
+		if deleteErr := s.dashboardStore.DeleteDashboard(ctx, &dashboards.DeleteDashboardCommand{
```

Comment:
can we try using a transaction and get rid of the manual deletion 🤞 (in the same or in a follow up PR)?

Reply:
As agreed, I'm doing it in a separate PR. Thanks!